pkg/experiment/query_backend/query_metrics.go

```diff
@@ -0,0 +1,238 @@
+package query_backend
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/grafana/dskit/runutil"
+
+	queryv1 "github.com/grafana/pyroscope/api/gen/proto/go/query/v1"
+	typesv1 "github.com/grafana/pyroscope/api/gen/proto/go/types/v1"
+	"github.com/grafana/pyroscope/pkg/experiment/query_backend/block"
+	phlaremodel "github.com/grafana/pyroscope/pkg/model"
+	parquetquery "github.com/grafana/pyroscope/pkg/phlaredb/query"
+	schemav1 "github.com/grafana/pyroscope/pkg/phlaredb/schemas/v1"
+)
+
+func init() {
+	registerQueryType(
+		queryv1.QueryType_QUERY_METRICS,
+		queryv1.ReportType_REPORT_METRICS,
+		queryMetrics,
+		newMetricsAggregator,
+		[]block.Section{
+			block.SectionProfiles,
+			block.SectionTSDB,
+			block.SectionSymbols,
+		}...,
+	)
+}
+
+func queryMetrics(q *queryContext, query *queryv1.Query) (r *queryv1.Report, err error) {
+	entries, err := profileEntryIterator(q)
+	if err != nil {
+		return nil, err
+	}
+	defer runutil.CloseWithErrCapture(&err, entries, "failed to close profile entry iterator")
+
+	var columns schemav1.SampleColumns
+	if err := columns.Resolve(q.ds.Profiles().Schema()); err != nil {
+		// TODO
+	}
+	column, err := schemav1.ResolveColumnByPath(q.ds.Profiles().Schema(), strings.Split("TotalValue", "."))
+	if err != nil {
+		return nil, err
+	}
+
+	rows := parquetquery.NewRepeatedRowIterator(q.ctx, entries, q.ds.Profiles().RowGroups(),
+		column.ColumnIndex,
+		columns.StacktraceID.ColumnIndex,
+		columns.Value.ColumnIndex)
+	defer runutil.CloseWithErrCapture(&err, rows, "failed to close column iterator")
+
+	labelsFromFp := make(map[uint64]phlaremodel.Labels)
+	builders := make(map[uint64]map[string]*phlaremodel.TimeSeriesBuilder)
+	for rows.Next() {
```

Comment:
I'm afraid this loop will suffer from very bad performance. Let's skip function filtering for now.